mypy/nodes.py

```diff
@@ -1101,6 +1111,19 @@ def __init__(self, expr: Optional[Expression]) -> None:
     def accept(self, visitor: StatementVisitor[T]) -> T:
         return visitor.visit_return_stmt(self)
 
+class ReturnSeeker(StatementVisitor):
+    def __init__(self) -> None:
+        self.found = False
+
+    def visit_statement(self, o: Statement) -> None:
+        if isinstance(o, ReturnStmt):
+            self.visit_return_stmt(o)
+
+    def visit_return_stmt(self, o: ReturnStmt) -> None:
+        if (o.expr is None or isinstance(o.expr, NameExpr) and o.expr.name == 'None'):
+            return
+        self.found = True
```

Comment:
This causes a bunch of code duplication with `stubgen.py`. Why don't you remove this class (and corresponding helper) from there and also use the added method there?